### PR TITLE
Updates for 0.12

### PR DESCRIPTION
**Pull request by kritzcreek**
Still in the process of figuring out if things actually work, need to update a few more things so I'd recommend holding off on merging until then.

### CODE REVIEW

**Comment by kritzcreek** [comment]
Okay, my manual tests indicate everything still works... so ready from me.

**Comment by nwolverson** [comment]
Hey not only does it typecheck, there are 15 passing unit tests which is about 15 more than most of my projects :D

File: src/Suggest.purs [hunk 1]
```diff
@@ -32,28 +32,23 @@ type Replacement =
   , replacement :: String
 }
 
-type SEff eff h = (console :: CONSOLE, exception :: EXCEPTION, fs :: FS, st :: ST h | eff)
-
--- This alias avoids psc bug which should be fixed in 9.0
-type PsaAnnotedErrors = Array PsaAnnotedError
```
Comment (nwolverson) [inline src/Suggest.purs:38]: I wonder if the typo was required to fix the bug :D

**Comment by kritzcreek** [comment]
Yeah I did run those, but I was mostly worried about the switch from STStrMap to `Ref (Map String )`

File: src/Suggest.purs (hunk 1 above)
Comment (kritzcreek) [reply src/Suggest.purs:38]: :D

**Pull request merged**